### PR TITLE
doc: explain supported os

### PR DESCRIPTION
**Pull request by ThibaultFy**
## Context and user need: 

Add the maintain and tested version (and degree of maintenance) for each OS and version of OS.

### CODE REVIEW

File: docs/source/operations/overview.rst [hunk 1]
```diff
@@ -23,6 +23,16 @@ For each component section below, configuration options relates to the component
 
 .. _cert-manager: https://cert-manager.io
 
+.. note::
```
Comment (RomainGoussault) [inline docs/source/operations/overview.rst:26]: Should be on the[ lib installation page](https://docs.substra.org/en/latest/substrafl_doc/substrafl_overview.html#installation) and not on the deployment page?

File: docs/source/operations/overview.rst [hunk 2]
```diff
@@ -23,6 +23,16 @@ For each component section below, configuration options relates to the component
...
+
+    We provide full support on the following OS:
```
Comment (RomainGoussault) [inline docs/source/operations/overview.rst:28]: I would say we provide support to every major OS (Ubuntu, MacOS, Windows). The difference is more on how deep we test them.

So we could rephrase:

> We provide full support on the following OS

By something like "the following OS version are deeply tested".


And for windows saying that only Windows Server 2022 is tested and only on subprocess modes. Howevers users should feel free to try other Windows version, and reach out on Slack if something goes wrong.

Comment (oleobal) [reply docs/source/operations/overview.rst:28]: We can't really provide much support on Windows due to not having Windows machines available though.

It's not a total blocker but if someone asks an Apple question we can answer a lot more rapidly (and probably better) than a Windows question

Comment (ThibaultFy) [reply docs/source/operations/overview.rst:28]: From my understanding, the idea here was to limit our promises on the support we can provide.

File: docs/source/substrafl_doc/substrafl_overview.rst [hunk 3]
```diff
@@ -25,7 +25,16 @@ Installation
...
 .. note::
-    On **Windows**, please refer to :ref:`local_mode` to use the Docker mode.
+
+    Full support on the following OS:
+        - **Ubuntu 22.06**
+        - **MacOS** (*Intel* or *M1* chip)
+
+    Support on Windows:
+        - ``subprocess`` mode is tested on **Windows Server 2022**.
+        - ``docker`` mode is not tested in the daily process. Please refer to :ref:`local_mode` to use it.
```
Comment (RomainGoussault) [inline docs/source/substrafl_doc/substrafl_overview.rst:35]:
```suggestion
        - ``docker`` mode is not tested for now. Please refer to :ref:`local_mode` to use it.
```

File: docs/source/substrafl_doc/substrafl_overview.rst [hunk 4]
```diff
@@ -25,7 +25,16 @@ Installation
...
+        - ``remote`` mode is not tested for now.
+
```
Comment (RomainGoussault) [inline docs/source/substrafl_doc/substrafl_overview.rst:37]: I would add something in the end:
Despite Windows not being deeply tested, fell free to [reach out](https://docs.substra.org/en/latest/additional/community.html) if you encounter any issue.

**Review by RomainGoussault — APPROVED** [review]
Thanks for the PR!

**Pull request merged**